structured/sql_api.proto

```diff
@@ -93,6 +93,6 @@ message SQLRequest {
 
 message SQLResponse {
   optional SQLResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
-  // The result set.
+  // The result set for the last Cmd in the request.
```

Comment:
That's fine; we just need to make sure that the list of columns is returned as a part of the repeated Result and not the header.